packages/numeric-input/src/NumericInput.js

```diff
@@ -0,0 +1,178 @@
+    } = this.props;
+
+    const numericInputStylesheet = (styles, props, themeData) => {
+      const numericInputStyles = customStylesheet(styles, props, themeData);
+      return stylesheet
+        ? stylesheet(numericInputStyles, props, themeData)
+        : numericInputStyles;
+    };
+
+    return (
+      <div>
+        <ControlBehavior
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onMouseDown={this.props.onMouseDown}
+          onMouseEnter={this.props.onMouseEnter}
+          onMouseLeave={this.props.onMouseLeave}
+          onMouseUp={this.props.onMouseUp}
+        >
+          {({
+            hasFocus,
+            hasHover,
+            isPressed,
+            onBlur: handleBlur,
+            onFocus: handleFocus,
+            onMouseDown: handleMouseDown,
+            onMouseEnter: handleMouseEnter,
+            onMouseLeave: handleMouseLeave,
+            onMouseUp: handleMouseUp
+          }) => (
+            <SpinnerBehavior
+              onClick={onClick}
+              value={value}
+              onChange={onChange}
+              initialValue={initialValue}
+              step={step}
+              disabled={disabled}
+            >
+              {({
+                handleClick,
+                handleChange,
+                increment,
+                decrement,
+                value: controlledValue,
+                setValue
+              }) => {
+                const onDirectChange = event => {
+                  const newValue = event.target.value;
+                  setValue(newValue);
+                };
+
+                return (
+                  <div style={{ position: "relative" }}>
+                    <SpinnerPresenter
+                      value={Number(value)}
+                      disabled={disabled}
+                      hasFocus={hasFocus}
+                      hasHover={hasHover}
+                      isPressed={isPressed}
+                      onBlur={handleBlur}
+                      onClick={handleClick}
+                      onChange={handleChange}
+                      onFocus={handleFocus}
+                      onMouseDown={handleMouseDown}
+                      onMouseEnter={handleMouseEnter}
+                      onMouseLeave={handleMouseLeave}
+                      onMouseUp={handleMouseUp}
+                      increment={increment}
+                      decrement={decrement}
+                      variant={variant}
+                      {...otherProps}
+                    />
+                    <Input
+                      {...otherProps}
+                      disabled={disabled}
+                      className="numeric-input"
+                      placeholder={placeholder}
+                      step={step}
+                      stylesheet={numericInputStylesheet}
+                      tagName="input"
+                      type="number"
+                      variant={variant}
+                      value={`${controlledValue}`}
```

Comment:
is this not the same as `value={controlledValue}`?

Reply:
I think controlledValue is the value passed down from SpinnerBehavior, so it's the updated value